From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Google <noreply@google.com>
Date: Thu, 30 Jan 2025 00:00:00 +0000
Subject: [PATCH] Limit memchr intrinsics to SSE2

Clang 21 on our build image lacks some of the newer AVX512 builtins
dragged in by `immintrin.internal.h`. memchr only needs the SSE2
primitives, so include the narrower header.
---
 libc/intrin/memchr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libc/intrin/memchr.c b/libc/intrin/memchr.c
index d27a4fdc..6c1f1a4b 100644
--- a/libc/intrin/memchr.c
+++ b/libc/intrin/memchr.c
@@ -15,7 +15,7 @@
 #include "libc/dce.h"
 #include "libc/nexgen32e/x86feature.h"
 #include "libc/str/str.h"
 #include "third_party/aarch64/arm_neon.internal.h"
-#include "third_party/intel/immintrin.internal.h"
+#include "third_party/intel/emmintrin.internal.h"
 #ifndef __aarch64__
 
 static inline const unsigned char *memchr_pure(const unsigned char *s,
-- 
2.34.1

